march_hardware/src/IMotionCube.cpp

```diff
@@ -76,41 +76,38 @@ void IMotionCube::validateMosiPDOs()
 void IMotionCube::writeInitialSettings(uint8 ecatCycleTime)
 {
   ROS_DEBUG("IMotionCube::writeInitialSettings");
-  bool success = true;
-  // sdo_bit32(slaveIndex, address, subindex, value);
 
   if (this->actuationMode == ActuationMode::unknown)
   {
     throw std::runtime_error("Cannot write initial settings to IMotionCube as it has actuation mode of unknown");
   }
 
   // mode of operation
-  success &= sdo_bit8(slaveIndex, 0x6060, 0, this->actuationMode.toModeNumber());
-
-  // position dimension index
-  success &= sdo_bit8(slaveIndex, 0x608A, 0, 1);
-
-  // position factor -- scaling factor numerator
-  success &= sdo_bit32(slaveIndex, 0x6093, 1, 1);
-  // position factor -- scaling factor denominator
-  success &= sdo_bit32(slaveIndex, 0x6093, 2, 1);
+  int mode_of_op = sdo_bit8(slaveIndex, 0x6060, 0, this->actuationMode.toModeNumber());
 
   // position limit -- min position
-  success &= sdo_bit32(slaveIndex, 0x607D, 1, this->encoder.getLowerSoftLimitIU());
+  int max_pos_lim = sdo_bit32(slaveIndex, 0x607D, 1, this->encoder.getLowerSoftLimitIU());
+
   // position limit -- max position
-  success &= sdo_bit32(slaveIndex, 0x607D, 2, this->encoder.getUpperSoftLimitIU());
+  int min_pos_lim = sdo_bit32(slaveIndex, 0x607D, 2, this->encoder.getUpperSoftLimitIU());
 
   // Quick stop option
-  success &= sdo_bit16(slaveIndex, 0x605A, 0, 6);
+  int stop_opt = sdo_bit16(slaveIndex, 0x605A, 0, 6);
 
   // Quick stop deceleration
-  success &= sdo_bit32(slaveIndex, 0x6085, 0, 0x7FFFFFFF);
+  int stop_decl = sdo_bit32(slaveIndex, 0x6085, 0, 0x7FFFFFFF);
+
+  // Abort connection option code
+  int abort_con = sdo_bit16(slaveIndex, 0x6007, 0, 1);
 
   // set the ethercat rate of encoder in form x*10^y
-  success &= sdo_bit8(slaveIndex, 0x60C2, 1, ecatCycleTime);
-  success &= sdo_bit8(slaveIndex, 0x60C2, 2, -3);
+  int rate_ec_x = sdo_bit8(slaveIndex, 0x60C2, 1, ecatCycleTime);
+  int rate_ec_y = sdo_bit8(slaveIndex, 0x60C2, 2, -3);
 
-  ROS_ASSERT_MSG(success, "Writing initial settings to IMC %d failed", this->slaveIndex);
+  if ((mode_of_op && max_pos_lim && min_pos_lim && stop_opt && stop_decl && abort_con && rate_ec_x && rate_ec_y) == 0)
```

Comment:
An `&&` operand will return a `bool`, however you check the result using `== 0`, which does not seem valid (but may work).
```suggestion
  if (mode_of_op && max_pos_lim && min_pos_lim && stop_opt && stop_decl && abort_con && rate_ec_x && rate_ec_y)
```